driver-async/src/main/com/mongodb/async/client/MongoCollection.java

```diff
@@ -156,15 +158,19 @@
      * Counts the number of documents in the collection.
      *
      * @param callback the callback passed the number of documents in the collection
+     * @deprecated use {@link #estimatedDocumentCount(SingleResultCallback)} or {@link #countDocuments(SingleResultCallback)} instead
      */
+    @Deprecated
     void count(SingleResultCallback<Long> callback);
 
     /**
      * Counts the number of documents in the collection according to the given options.
      *
      * @param filter   the query filter
      * @param callback the callback passed the number of documents in the collection
+     * @deprecated use {@link #countDocuments(Bson, SingleResultCallback)} instead
```

Comment:
I have a slight preference for the first doc comment just above, which mentions both of the new methods